protocol/binary/reader.go

```diff
@@ -261,6 +261,8 @@ func (br *Reader) readBytes(off int64) ([]byte, int64, error) {
 	// bytesAllocThreshold. We don't want bad requests to lock the system up.
 	if length > bytesAllocThreshold {
 		var buff bytes.Buffer
+		buff.Grow(int(length))
```

Comment:
See comment right above this block.
This is a slow path specifically for requests larger than 1MB.
Requests smaller than 1MB will be pre-allocated as desired.

Reply:
Not resolved. Comment still valid.